app/controllers/application_controller.rb

```diff
@@ -72,7 +72,7 @@ def customize_params
   # the user to after a successful log in
   def after_sign_in_path_for(resource)
     if current_user.saw_onboarding
-      path = request.env["omniauth.origin"] || stored_location_for(resource) || dashboard_path
+      path = stored_location_for(resource) || request.env["omniauth.origin"] || root_path(signin: "true")
```

Comment:
Moved up in priority (if available) the use of Devise's `stored_location_for` for redirecting back to the correct location

Reply:
I think we lost some functionality now. As a logged out user check https://dev.to/new and http://localhost:3000/new (I actually think we should protect a route like /new and then redirect to it once logged in)

Reply:
Thank you for this @rhymes! That's true, the `localhost:3000/new` used to redirect directly to the editor as a follow up from a logged-out user clicking on "Write a post".

I think we were pushing for a unified CTA in the navigation bar to make it easier (fewer buttons) for users to register/login. So right now I don't think there's a way to get to `/new` unless you type it directly in the URL, right?

Asking @lisasy @benhalpern about their thoughts on this because we can either remove unused logic (from [this if-else](https://github.com/forem/forem/blob/master/app/views/devise/registrations/_registration_form.html.erb#L2-L15)) or I can add the necessary changes to make sure we redirect to the editor correctly after logging in from `/new`

Reply:
Sure, there's no link to it right now, but people might have boomarked the link or end up going there while typing on the address bar of the browser. I wonder what happens if you submit an article while you're logged out :eyes:

Reply:
I just pushed a change that makes sure the redirect (after signing in/signing up) from `/new` goes to back the editor to avoid loosing the functionality.

I think the conversation around whether or not keeping this in the long run is valid but maybe good to have it elsewhere and ship this PR if it looks good for now 🙂